lib.rs

```diff
@@ -93,3 +93,846 @@ pub mod src {
+            self.dav1d_settings.max_frame_delay = max_frame_delay as i32;
+        }
+
+        pub fn get_max_frame_delay(&self) -> u32 {
+            self.dav1d_settings.max_frame_delay as u32
+        }
+
+        pub fn set_apply_grain(&mut self, apply_grain: bool) {
+            self.dav1d_settings.apply_grain = i32::from(apply_grain);
+        }
+
+        pub fn get_apply_grain(&self) -> bool {
+            self.dav1d_settings.apply_grain != 0
+        }
+
+        pub fn set_operating_point(&mut self, operating_point: u32) {
+            self.dav1d_settings.operating_point = operating_point as i32;
+        }
+
+        pub fn get_operating_point(&self) -> u32 {
+            self.dav1d_settings.operating_point as u32
+        }
+
+        pub fn set_all_layers(&mut self, all_layers: bool) {
+            self.dav1d_settings.all_layers = i32::from(all_layers);
+        }
+
+        pub fn get_all_layers(&self) -> bool {
+            self.dav1d_settings.all_layers != 0
+        }
+
+        pub fn set_frame_size_limit(&mut self, frame_size_limit: u32) {
+            self.dav1d_settings.frame_size_limit = frame_size_limit;
+        }
+
+        pub fn get_frame_size_limit(&self) -> u32 {
+            self.dav1d_settings.frame_size_limit
+        }
+
+        pub fn set_strict_std_compliance(&mut self, strict_std_compliance: bool) {
+            self.dav1d_settings.strict_std_compliance = i32::from(strict_std_compliance);
+        }
+
+        pub fn get_strict_std_compliance(&self) -> bool {
+            self.dav1d_settings.strict_std_compliance != 0
+        }
+
+        pub fn set_output_invisible_frames(&mut self, output_invisible_frames: bool) {
+            self.dav1d_settings.output_invisible_frames = i32::from(output_invisible_frames);
+        }
+
+        pub fn get_output_invisible_frames(&self) -> bool {
+            self.dav1d_settings.output_invisible_frames != 0
+        }
+
+        pub fn set_inloop_filters(&mut self, inloop_filters: InloopFilterType) {
+            self.dav1d_settings.inloop_filters = inloop_filters.bits();
+        }
+
+        pub fn get_inloop_filters(&self) -> InloopFilterType {
+            InloopFilterType::from_bits_truncate(self.dav1d_settings.inloop_filters)
+        }
+
+        pub fn set_decode_frame_type(&mut self, decode_frame_type: DecodeFrameType) {
+            self.dav1d_settings.decode_frame_type = decode_frame_type.into();
+        }
+
+        pub fn get_decode_frame_type(&self) -> DecodeFrameType {
+            DecodeFrameType::try_from(self.dav1d_settings.decode_frame_type)
+                .expect("Invalid Dav1dDecodeFrameType")
+        }
+    }
+
+    bitflags::bitflags! {
+        #[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+        pub struct InloopFilterType: u32 {
+            const DEBLOCK = DAV1D_INLOOPFILTER_DEBLOCK;
+            const CDEF = DAV1D_INLOOPFILTER_CDEF;
+            const RESTORATION = DAV1D_INLOOPFILTER_RESTORATION;
+        }
+    }
```

Comment:
We'd prefer just making `Rav1dInloopFilterType` this instead of doing a conversion here at the edge of the API.